packages/query/src/helpers/identify-txs.ts

```diff
@@ -147,28 +159,48 @@ const generateTxId = async (tx: Transaction): Promise<TransactionId> => {
   return new TransactionId({ inner: await sha256Hash(tx.toBinary()) });
 };
 
+const isSpendableNoteRecord = (
+  note: SpendableNoteRecord | SwapRecord,
+): note is SpendableNoteRecord => {
+  return note.getType().typeName === SpendableNoteRecord.typeName;
+};
+const getAddressIndexFromNote = (
+  note: SpendableNoteRecord | SwapRecord,
+): AddressIndex | undefined => {
+  if (isSpendableNoteRecord(note)) {
+    return note.addressIndex;
+  }
+  return undefined;
+};
+
 const searchRelevant = async (
   tx: Transaction,
-  spentNullifiers: Set<Nullifier>,
+  spentNullifiers: Map<Nullifier, SpendableNoteRecord | SwapRecord>,
   commitmentRecords: Map<StateCommitment, SpendableNoteRecord | SwapRecord>,
-  isControlledAddr: ViewServerInterface['isControlledAddress'],
+  getIndexByAddress: ViewServerInterface['getIndexByAddress'],
 ): Promise<
   { relevantTx: RelevantTx; recoveredSourceRecords: RecoveredSourceRecords } | undefined
 > => {
   let txId: TransactionId | undefined; // If set, that means this tx is relevant and should be returned to the caller
+  let subaccount: AddressIndex | undefined;
```

Comment:
this change collects subaccounts from `SpendableNoteRecord`s but there's no way to get AddressIndex for all possible actions, that's why it might be `undefined`. However, it is enough to get them for the purpose of the tasks – to identify LPs per subaccount